pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -238,6 +240,7 @@
 
     private final ReentrantLock mutex = new ReentrantLock();
     private final Condition isClosedCondition = mutex.newCondition();
+    private final AtomicReference<CompletableFuture<Void>> closeFutureReference = new AtomicReference<>();
```

Comment:
Since we're not doing any CAS operations (just get and set), we don't the `AtomicReference`, but just makie the variable `volatile`.

Reply:
done